### PR TITLE
updates to default app and cli usage

### PR DESCRIPTION
**Pull request by codebytere**
Tracks progress on https://github.com/electron/electron/issues/5345

Tasks:
- [x] Make help response the response for no args specified
- [x] Make default app a flagged option
- [x] show Chrome, Node, and V8 versions
- [x] link to `electron-forge`
- [x] remove references to prebuilt 
- [x] link the logo to the website
- [x] Add more information to `default-app`, maybe some links to resources and examples? 

Please feel free to add input!

Thought it could use some design sprucing:

screen shot 2017-09-27 at 11 56 18 pm

(I'm aware of some of the spacing/padding issues and am working on them)

/cc @zeke

### CODE REVIEW

File: default_app/index.html [hunk 1]
```diff
@@ -112,62 +136,93 @@
...
+        </a>
+      </div>
+      <div class="linkcol">
+        <h4>API Demos</h4>
+        <a class="hero-link" href="https://github.com/electron/electron-api-demos">
+          <span class="octicon hero-octicon octicon-star" style="padding-left:2px" aria-hidden="true"></span>
+        </a>
+      </div>
+      <div class="linkcol">
+        <h4>Electron-Forge</h4>
```
Comment (malept) [inline default_app/index.html]: I'd prefer either "Electron Forge" or "electron-forge", not the hybrid 😁

File: default_app/index.html [hunk 2]
```diff
@@ -112,62 +136,93 @@
...
+        <a class="hero-link" href="https://electronforge.io">
+          <span class="octicon hero-octicon octicon-telescope" style="padding-left:2px" aria-hidden="true"></span>
```
Comment (malept) [inline default_app/index.html]: Too bad there isn't an anvil octicon 😄

Comment (codebytere) [reply default_app/index.html]: if you'd prefer a different one, let me know!

Comment (malept) [reply default_app/index.html]: @MarshallOfSound will have an opinion on this

**Comment by zeke** [comment]
I took a shot at improving the styles, plus a few other changes:

- Changed `Chrome` to `Chromium`
- Made link text clickable
- Added an unselectable `$` prompt to make it more obvious that it's a command
- Fix link handler, which was broken by links being inside `<span>` tags
- Add `utm_source` query param to outbound links so we can track default_app link clicks.

Here's what it's looking like now:

screen shot 2017-09-29 at 12 13 02 pm

File: default_app/renderer.js [hunk 3]
```diff
@@ -1,45 +1,23 @@
...
-const version = process.versions.electron
-document.querySelector('.header-version').innerText = version
-document.querySelector('.command-example').innerText = `${execPath} path-to-your-app`
-document.querySelector('.quick-start-link').href = `https://github.com/electron/electron/blob/v${version}/docs/tutorial/quick-start.md`
-document.querySelector('.docs-link').href = `https://github.com/electron/electron/tree/v${version}/docs#readme`
+const path = require('path')
+const URL = require('url')
+const electronPath = path.relative(process.cwd(), remote.process.execPath)
+
+Array.from(document.querySelectorAll('a[href]')).forEach(link => {
```
Comment (felixrieseberg) [inline default_app/renderer.js]: I got a minor preference for `[...document.querySelectorAll('a[href]')], but it's the same thing, so whatever you feel like!

File: default_app/renderer.js [hunk 4]
```diff
@@ -1,45 +1,23 @@
...
+    e.preventDefault()
+    shell.openExternal(url)
+  })
+})
+
+document.querySelector('.electron-version').innerText = `Electron v${process.versions.electron}`
+document.querySelector('.chrome-version').innerText = `Chromium v${process.versions.chrome}`
+document.querySelector('.node-version').innerText = `Node v${process.versions.node}`
+document.querySelector('.v8-version').innerText = `v8 v${process.versions.v8}`
+
```
Comment (felixrieseberg) [inline default_app/renderer.js]: Why this newline?

Comment (codebytere) [reply default_app/renderer.js]: ...unclear....
will change, whoops

**Comment by codebytere** [comment]
won't be implemented for now :D

**Comment by zeke** [comment]
Glad to see this shipping. Curious about the deprecation notice, though:

>  Deprecation Warning: To render the default app, the -d or --default flags will soon need to be used.

By the time a user is reading this, doesn't the `-d` flag need to be used?

**Comment by codebytere** [comment]
hmm, maybe i can word this more clearly? I changed it (following a chat with @felixrieseberg) such that the default app still renders with no args, as well as with new `-d` and `--default` flags, and told users that the default rendering would soon disappear so that there was a transition into the breaking change. Is there a better warning you think might convey that line of thinking?

**Comment by MarshallOfSound** [comment]
> Deprecation Warning: To render the default app, the -d or --default flags will soon need to be used.

I actually didn't see this, with this change it means users simply launching "Electron.app" will not see anything, I think we should always show the default app like we currently do.  Is there a reason / discussion about that change in behavior?

**Comment by MarshallOfSound** [comment]
> so that there was a transition into the breaking change. 

FWIW just building current master and launching the output from finder results in the behavior I described above, the app icon appears briefly and then the app closes itself

**Comment by codebytere** [comment]
hmm, that wasn't happening when i was testing locally 🤔 i discussed it with @zeke originally, just because i felt that the help information should be more of a 1st class citizen and not only available via `--help` flag. At the time it seemed like the best move was to move over default app and replace it with help text, but i dont see any reason we can't keep both, now that i think about it. what are y'alls thoughts?

**Comment by codebytere** [comment]
Pushed that change to a branch: https://github.com/electron/electron/tree/remove-default-deprecation 
 
Fixed the app close, an exit snuck in there :(

**Pull request merged**